### ISSUE TITLE
Preservation of Fine Details

### ISSUE THREAD

**Issue by TwilightArchon** [nagejacob/SpatiallyAdaptiveSSID#13]
Thanks for your work! After training the network on my own dataset, the noise are removed very well. 
However, I found that the places without noise are also denoised, which results in the loss of fine details. May I ask what is a good way to preserve the details? Would adding the structural loss and perception loss between the noised image and denoised image help?

**Comment by nagejacob** [comment]
Can you show me an example of your noisy/denoised pairs? Besides, do you perform all the three training stages? To be honest, self-supervised image denoising methods for real world images often suffers from loss of details. We have tried to solve this problem but not achieve good results yet.

**Comment by TwilightArchon** [comment]
Of course! The first image is the real world image, and the second one is the denoised one. There is some color distortion, but I think it can be solved by adding some loss function related to color. I'm currently training with structure loss function added to the  BNN and LAN networks, hoping it can have some effects. Also, do you think that adding images with no noise during training would help?
![image](https://github.com/nagejacob/SpatiallyAdaptiveSSID/assets/134197144/6d87bafd-c434-4109-b5c8-68a2b45a530c)
![image](https://github.com/nagejacob/SpatiallyAdaptiveSSID/assets/134197144/b6efbc23-509b-4530-b68e-d98b559dfa76)

**Comment by nagejacob** [comment]
As BNN adopts large blind-neighborhood, it could not recover the texture details like this. Although LAN could help to recover some texture, it is still hard to preserve fine-grained details. I think adding some paired datasets (e.g., SIDD) when distilling U-Net may help, that is to train U-Net with paired data generated from BNN and LAN, and paired data in SIDD. For simplicity, just distill U-Net with BNN outputs and paired data in SIDD.

**Comment by TwilightArchon** [comment]
Do you mean first training on my custom dataset, and after that, train the Unet component on the SIDD dataset, calculating loss between the denoised and groundtruth to improve the model's ability to generate fine details?

**Comment by nagejacob** [comment]
No, use both your custom dataset and SIDD for training the U-Net, like a half custom data and a half SIDD data in a batch.

**Comment by TwilightArchon** [comment]
SIDD is paired with ground truth, do you mean training with half noisy images from custom dataset, and the other half from SIDD noisy dataset might, to some extent, improve the model's performance on the images without noise? And the ground truth of SIDD is not used, since we only use the output of BNN and LAN to supervise the Unet's training right?

**Comment by nagejacob** [comment]
No, when you train U-Net, you use {your_noisy, BNN_result} and {your_noisy, LAN_result} as dataset. Now mix {your_noisy, BNN_result}, {your_noisy, LAN_result} and {SIDD_noisy, SIDD_gt} as dataset. The gt in SIDD paired data is sharp so it may improve the details of U-Net.

**Comment by TwilightArchon** [comment]
Got it, thanks!
